packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -25,8 +25,10 @@ const String _kPath = '/runner';
 
 String shellPath;
 
-void installHook() {
-  hack.registerPlatformPlugin(<TestPlatform>[TestPlatform.vm], () => new FlutterPlatform());
+void installHook({ int observatoryPort }) {
+  hack.registerPlatformPlugin(
+      <TestPlatform>[TestPlatform.vm],
+      () => new FlutterPlatform(observatoryPort: observatoryPort));
 }
```

Comment:
Could do! I'm using the existing installHook() internally, so I'd need to replace that with a re-implementation that sets the observatory port if we don't do it here.